### PR TITLE
CFE-2698: Replaced CFE-2698 acceptance test with 3 separate tests for unless

### PR DESCRIPTION
**Pull request by olehermanse**

### CODE REVIEW

**Comment by olehermanse** [comment]
@cf-bottom jenkins

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/2798/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/2798/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-2798/

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/2799/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/2799/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-2799/

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/2803/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/2803/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-2803/

**Comment by olehermanse** [comment]
@cf-bottom let's try one more time in jenkins. Looks like a spurious error in deployment tests.

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/2805/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/2805/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-2805/

File: tests/acceptance/00_basics/04_bundles/unless_bug.cf [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        or => {
+            isvariable("test.vars_promise_which_should_be_skipped"),
+        };
+
+  reports:
+    # TOOD: Swap these when bug is fixed:
+    not_ok::
+      "$(this.promise_filename) Pass";
+    !not_ok::
+      "$(this.promise_filename) FAIL";
```
Comment (vpodzime) [inline tests/acceptance/00_basics/04_bundles/unless_bug.cf]: This should use the `test_soft_fail` mechanism.

Comment (olehermanse) [reply tests/acceptance/00_basics/04_bundles/unless_bug.cf]: https://github.com/cfengine/core/pull/3852

**Pull request merged**